--- routes/inbox.js.orig	1985-10-26 09:15:00.000000000 +0100
+++ routes/inbox.js	2019-06-07 22:15:32.371416369 +0100
@@ -4,7 +4,8 @@
       request = require('request'),
       router = express.Router();

-function signAndSend(message, name, domain, req, res, targetDomain) {
+function signAndSend(message, name, domain, req, res, targetDomain) {
+  message = message.replace(/https/g, 'http');
   // get the URI of the actor object and append 'inbox' to it
   let inbox = message.object.actor+'/inbox';
   let inboxFragment = inbox.replace('https://'+targetDomain,'');
@@ -23,7 +24,8 @@
     signer.end();
     const signature = signer.sign(privkey);
     const signature_b64 = signature.toString('base64');
-    let header = `keyId="https://${domain}/u/${name}",headers="(request-target) host date",signature="${signature_b64}"`;
+    let header = `keyId="http://${domain}/u/${name}",headers="(request-target) host date",signature="${signature_b64}"`;
+    console.log('header', header);
     request({
       url: inbox,
       headers: {
@@ -69,8 +71,9 @@
 router.post('/', function (req, res) {
   // pass in a name for an account, if the account doesn't exist, create it!
   let domain = req.app.get('domain');
+  console.log(req.body);
   const myURL = new URL(req.body.actor);
-  let targetDomain = myURL.hostname;
+  let targetDomain = myURL.host;
   // TODO: add "Undo" follow event
   if (typeof req.body.object === 'string' && req.body.type === 'Follow') {
     let name = req.body.object.replace(`https://${domain}/u/`,'');
